### PR TITLE
API reserved parameter language and parameter format clarifications

### PR DESCRIPTION
**Pull request by philvarner**
API clarifications: 
* reserved parameter language and parameter format clarifications 
* GET vs. POST JSON parameters

I split these changes out of https://github.com/radiantearth/stac-spec/pull/500

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all`. to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

**Comment by m-mohr** [comment]
**PR Checklist:**

- [ ] This PR has breaking changes.
- [ ] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md).
- [ ] API only: I have run `npm run generate-all`. to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

File: api-spec/api-spec.md [hunk 1]
```diff
@@ -64,4 +66,41 @@ Unless otherwise noted by **Path-only**, these filters are passed as query strin
 | ids | [string] | Array of Item ids to return. All other filter parameters that further restrict the number of search results (except `next` and `limit`) are ignored |
 | collections  | [string]         | STAC       | Array of Collection IDs to include in the search for items. Only Items in one of the provided Collections will be searched |
 
-Only one of either **intersects** or **bbox** should be specified.  If both are specified, a Bad Request response should be returned. 
\ No newline at end of file
+Only one of either **intersects** or **bbox** should be specified.  If both are specified, a Bad Request response should be returned.
```
Comment (m-mohr) [inline api-spec/api-spec.md]: Adding 400 before "Bad Request" (as below) would make sense, I think.

File: api-spec/extensions/query/README.md [hunk 2]
```diff
@@ -44,3 +44,28 @@ Find scenes with cloud cover between 0 and 10%:
   }
 }
 ```
+
+## Design Note
```
Comment (matthewhanson) [inline api-spec/extensions/query/README.md]: I don't think this design note is necessary, it's just providing a justification for why this wasn't done some other way, and don't think we should be in the habit of feeling like we have to do so.  I've recently removed other similar design notes from other parts of the spec.

Comment (m-mohr) [reply api-spec/extensions/query/README.md]: Agreed.

File: api-spec/api-spec.md [hunk 3]
```diff
@@ -64,4 +66,41 @@ Unless otherwise noted by **Path-only**, these filters are passed as query strin
...
+ implemented by an API, then if that parameter has a non-empty value in the request a 400 Bad Request status code must 
+ be returned. 
+ 
+### Fields Extension
+ 
+These parameters and fields are reserved for the Fields extension.
+ 
+| Parameter | Type              | APIs       | Description |
+| --------- | ----------------- | ---------- | ----------- |
+| fields    | string or [Field] | Placeholder parameter for [API Fields Extension](extensions/fields/README.md). |
```
Comment (m-mohr) [inline api-spec/api-spec.md]: We usually seperate data types with a pipe, e.g. `string|[Field]`

File: api-spec/extensions/query/README.md (hunk 2 above)
Comment (philvarner) [reply api-spec/extensions/query/README.md]: removed.

File: api-spec/api-spec.md (hunk 1 above)
Comment (philvarner) [reply api-spec/api-spec.md]: done.

**Comment by philvarner** [comment]
All recommended changes have been made.

**Pull request merged**